From 7c350e0f504bb96bf99978bed48cfbd41deafda2 Mon Sep 17 00:00:00 2001
From: Vinaykumar Masane <vinaykumar.masane@freescale.com>
Date: Thu, 26 Nov 2015 16:13:50 +0530
Subject: [PATCH 25/60] SBR-215 Removed sw_ver field from boardinfo

SW version will be referenced from Active FM image version from Linux

Signed-off-by: Vinaykumar Masane <vinaykumar.masane@freescale.com>
---
 commands/boardinfo.c | 144 ++++++++++++++++++---------------------------------
 1 file changed, 51 insertions(+), 93 deletions(-)

diff --git a/commands/boardinfo.c b/commands/boardinfo.c
index c7a9719..8f344c0 100644
--- a/commands/boardinfo.c
+++ b/commands/boardinfo.c
@@ -40,7 +40,6 @@
 #define SIZE_VID  3         /* Max  03 Character H/W Version */
 #define SIZE_S_N 11         /* Max  11 Character Board Serial Number */
 #define SIZE_P_N 12         /* Max  12 Character Board Part Number */
-#define SIZE_SW_VER 4       /* Max  04 Character Board Software Version */
 #define SIZE_PROD_NAME 16   /* Max  16 Character Board Product Name */
 #define SIZE_PROD_SERIES 16 /* Max  16 Character Board Product Series */
 #define SIZE_SW_DESC 128    /* Max 128 Character Board Software Description */
@@ -60,7 +59,6 @@ typedef struct c2krv340_board_info {
   unsigned char s_n[16];             /* Board Serial Number */
   unsigned char mac[3][8];           /* MAC Information */
   unsigned char p_n[16];             /* Board Serial Number */
-  unsigned char sw_ver[16];          /* Board Sowftware Version */
   unsigned char prod_name[64];       /* Board Product Name */
   unsigned char prod_series[16];     /* Board Product Series */
   unsigned char sw_desc[128];        /* Board Software Description */
@@ -77,7 +75,7 @@ static int command_boardinfo_mac(int argc, char *argv[])
   c2krv340 *board_data = NULL;
   unsigned char macdata[512];
   int mac_offset,j;
-  int count=2;
+  int count = 2;
   fd = open(filename, O_RDONLY, 0);
   if (fd < 0) {
     printf("could not open %s!\n", filename);
@@ -131,16 +129,16 @@ static int command_boardinfo_mac(int argc, char *argv[])
   }
 
   count=count + mac_offset;
-  mac_offset = macdata[count+1];
-  count = count + mac_offset+2;
-  mac_offset = macdata[count+1];
+  mac_offset = macdata[count + 1];
+  count = count + mac_offset + 2;
+  mac_offset = macdata[count + 1];
 
   count = count + mac_offset + 2;
   for (i = 0; i < COUNT_MAC; i++)
   {
     memset(ifname, 0x00, sizeof(ifname));
     memset(out_str, 0x00, sizeof(out_str));
-    count = count+2;
+    count = count + 2;
     for(j=0;j<6;j++)
        board_data->mac[i][j] = macdata[count++];
 
@@ -167,7 +165,7 @@ static int command_boardinfo_update(int argc, char *argv[])
   int i, j, len = 2, n_mac_octet = 0;
   char c = '\0';
   unsigned char type = 0xFF;
-  unsigned int count=2;
+  unsigned int count = 2;
   unsigned char boarddata[512] = {'\0'};
   unsigned char macdata[3][6];
   enum mac_octet_state {
@@ -207,9 +205,9 @@ static int command_boardinfo_update(int argc, char *argv[])
   boarddata[count - len] = type;
   boarddata[count - len + 1] = len - 2;
 
-  len = 2;
-  count = count+len;
   printf("\nEnter VID : ");
+  len = 2;
+  count = count + len;
   while (len < (SIZE_VID + 2))
   {
     c = getc();
@@ -234,10 +232,10 @@ static int command_boardinfo_update(int argc, char *argv[])
   boarddata[count - len] = type;
   boarddata[count - len + 1] = len - 2;
 
-  len = 2;
   printf("\nEnter S/N : ");
-  count = count+len;
-  while (len < (SIZE_S_N + 2 ))
+  len = 2;
+  count = count + len;
+  while (len < (SIZE_S_N + 2))
   {
     c = getc();
     if ( isalnum(c) || c == KEY_CR )
@@ -259,7 +257,7 @@ static int command_boardinfo_update(int argc, char *argv[])
   }
     type = 0xC1;
     boarddata[count - len ] = type;
-    boarddata[count - len +1] = len-2;
+    boarddata[count - len + 1] = len - 2;
 
   for (i = 0; i < COUNT_MAC; i++)
   {
@@ -305,8 +303,8 @@ static int command_boardinfo_update(int argc, char *argv[])
   }
 
   printf("\nEnter Part No : ");
-  len=2;
-  count=count+len;
+  len = 2;
+  count = count + len;
   while (len < (SIZE_P_N + 2))
   {
     c = getc();
@@ -330,41 +328,11 @@ static int command_boardinfo_update(int argc, char *argv[])
    }
   type = 0xC0;
   boarddata[count - len ] = type;
-  boarddata[count - len +1] = len-2;
-
-  printf("\nEnter SW Version : ");
-  len=2;
-  count=count+len;
-  while (len < (SIZE_SW_VER + 2))
-  {
-    c = getc();
-    if (c == KEY_CR)
-      break;
-
-    if ( isprint(c))
-    {
-      c = toupper(c);
-      putchar(c);
-    }
-    else
-    {
-      printf("\n c = %c : 0x%X Input Error! %s", c, c, help_string);
-      return COMMAND_ERROR;
-    }
-    if ( c == KEY_CR )
-    {
-      break;
-    }
-    len++;
-    boarddata[count++] = c;
-  }
-  type = 0x8C;
-  boarddata[count - len ] = type;
-  boarddata[count - len +1] = len-2;
+  boarddata[count - len + 1] = len - 2;
 
   printf("\nEnter PROD Name : ");
-  len=2;
-  count=count+len;
+  len = 2;
+  count = count + len;
   while (len < (SIZE_PROD_NAME + 2))
   {
     c = getc();
@@ -390,11 +358,11 @@ static int command_boardinfo_update(int argc, char *argv[])
   }
   type = 0xDB;
   boarddata[count - len ] = type;
-  boarddata[count - len +1] = len-2;
+  boarddata[count - len + 1] = len - 2;
 
   printf("\nEnter PROD Series : ");
-  len=2;
-  count=count+len;
+  len = 2;
+  count = count + len;
   while (len < (SIZE_PROD_SERIES + 2))
   {
     c = getc();
@@ -420,11 +388,11 @@ static int command_boardinfo_update(int argc, char *argv[])
   }
   type = 0xA0;
   boarddata[count - len ] = type;
-  boarddata[count - len +1] = len-2;
+  boarddata[count - len + 1] = len - 2;
 
   printf("\nEnter SOFTWARE Description : ");
-  len=2;
-  count=count+len;
+  len = 2;
+  count = count + len;
   while (len < (SIZE_SW_DESC + 2))
   {
     c = getc();
@@ -450,11 +418,11 @@ static int command_boardinfo_update(int argc, char *argv[])
   }
   type = 0xA1;
   boarddata[count - len ] = type;
-  boarddata[count - len +1] = len-2;
+  boarddata[count - len + 1] = len - 2;
 
   printf("\nEnter SOFTWARE Objectid : ");
-  len=2;
-  count=count+len;
+  len = 2;
+  count = count + len;
   while (len < (SIZE_SW_OBJID + 2))
   {
     c = getc();
@@ -480,7 +448,7 @@ static int command_boardinfo_update(int argc, char *argv[])
   }
   type = 0xA2;
   boarddata[count - len ] = type;
-  boarddata[count - len +1] = len-2;
+  boarddata[count - len + 1] = len - 2;
 
   fd = open(filename, O_RDWR, 0);
   if (fd < 0) {
@@ -510,7 +478,7 @@ static int command_boardinfo_update(int argc, char *argv[])
 
 static int command_boardinfo_show(int argc, char *argv[])
 {
-  int fd = -1, n_read = 0, i = 0, count=2;
+  int fd = -1, n_read = 0, i = 0, count = 2;
   char out_str[18] = {0}, ifname[6] = {0};
   c2krv340 *board_data = NULL;
   char buf_boarddata[512];
@@ -552,7 +520,7 @@ static int command_boardinfo_show(int argc, char *argv[])
   }
 
   printf("\nPID  : ");
-  for (i=0;i<pid_length;i++)
+  for (i = 0; i < pid_length; i++)
   {
     if (isprint(buf_boarddata[count]))
       putchar(buf_boarddata[count]);
@@ -561,10 +529,10 @@ static int command_boardinfo_show(int argc, char *argv[])
     count++;
   }
 
-  vid_length = buf_boarddata [count+1];
-  count = count+2;
+  vid_length = buf_boarddata[count + 1];
+  count = count + 2;
   printf("\nVID  : ");
-  for (i=0;i<vid_length;i++)
+  for (i = 0; i < vid_length; i++)
   {
     if (isprint(buf_boarddata[count]))
       putchar(buf_boarddata[count]);
@@ -573,10 +541,10 @@ static int command_boardinfo_show(int argc, char *argv[])
     count++;
   }
 
-  sn_length = buf_boarddata [count+1];
-  count = count+2;
+  sn_length = buf_boarddata[count + 1];
+  count = count + 2;
   printf("\nS/N  : ");
-  for (i=0;i<sn_length;i++)
+  for (i = 0; i < sn_length; i++)
   {
     if (isprint(buf_boarddata[count]))
       putchar(buf_boarddata[count]);
@@ -584,11 +552,12 @@ static int command_boardinfo_show(int argc, char *argv[])
       putchar(' ');
     count++;
   }
+
   for (i = 0; i < COUNT_MAC; i++)
   {
-    mac_length=buf_boarddata[count+1];
-    count = count+2;
-    for (j=0;j<mac_length;j++)
+    mac_length=buf_boarddata[count + 1];
+    count = count + 2;
+    for (j = 0; j < mac_length; j++)
     {
       mac_data[i][j] = buf_boarddata[count++];
     }
@@ -606,22 +575,11 @@ static int command_boardinfo_show(int argc, char *argv[])
     printf("\n%-4s MAC : %s", ifname, out_str);
     setenv(ifname, out_str);
   }
-  pn_length = buf_boarddata [count+1];
-  count = count+2;
-  printf("\nP/N  : ");
-  for (i=0;i<pn_length;i++)
-  {
-    if (isprint(buf_boarddata[count]))
-      putchar(buf_boarddata[count]);
-    else
-      putchar(' ');
-    count++;
-  }
 
-  swver_length = buf_boarddata [count+1];
-  count = count+2;
-  printf("\nSW_VER  : ");
-  for (i = 0; i < swver_length; i++)
+  pn_length = buf_boarddata[count + 1];
+  count = count + 2;
+  printf("\nP/N  : ");
+  for (i = 0; i < pn_length; i++)
   {
     if (isprint(buf_boarddata[count]))
       putchar(buf_boarddata[count]);
@@ -630,8 +588,8 @@ static int command_boardinfo_show(int argc, char *argv[])
     count++;
   }
 
-  prodname_length = buf_boarddata [count+1];
-  count = count+2;
+  prodname_length = buf_boarddata[count + 1];
+  count = count + 2;
   printf("\nPROD_NAME  : ");
   for (i = 0; i < prodname_length; i++)
   {
@@ -642,8 +600,8 @@ static int command_boardinfo_show(int argc, char *argv[])
     count++;
   }
 
-  prodsrs_length = buf_boarddata [count+1];
-  count = count+2;
+  prodsrs_length = buf_boarddata[count + 1];
+  count = count + 2;
   printf("\nPROD_SERIES  : ");
   for (i = 0; i < prodsrs_length; i++)
   {
@@ -654,8 +612,8 @@ static int command_boardinfo_show(int argc, char *argv[])
     count++;
   }
 
-  swdesc_length = buf_boarddata [count+1];
-  count = count+2;
+  swdesc_length = buf_boarddata[count + 1];
+  count = count + 2;
   printf("\nSW_DESC  : ");
   for (i = 0; i < swdesc_length; i++)
   {
@@ -666,8 +624,8 @@ static int command_boardinfo_show(int argc, char *argv[])
     count++;
   }
 
-  swobjid_length = buf_boarddata [count+1];
-  count = count+2;
+  swobjid_length = buf_boarddata[count + 1];
+  count = count + 2;
   printf("\nSW_OBJID  : ");
   for (i = 0; i < swobjid_length; i++)
   {
-- 
2.7.4

